### ISSUE TITLE
The 'dragstart' event is fired globally for all Drag components - bug or intended?

### ISSUE THREAD

**Issue by ddenev** [rlemaigre/Easy-DnD#37]
I noticed that the `dragstart` event on a Drag component is emitted for all Drag components in the application, it is not emitted per component but rather globally.

What I mean is:

Imagine we have 3 Drag components registered anywhere in the application (no need to be in the same parent component) and all these 3 components have `@dragstart` listeners attached to them, e.g.:

```
...
<drag ref="one" @dragstart="handleDragStart1" />
...
<drag ref="two" @dragstart="handleDragStart2" />
...
<drag ref="three" @dragstart="handleDragStart3" />
...
```
Whenever I drag any of them (for example, 'two') all 3 drag components receive the `dragstart` event and all 3 event handlers are  called (`handleDragStart1`, `handleDragStart2` and `handleDragStart3`).

https://codesandbox.io/s/naughty-pascal-rjioj?file=/src/App.vue

**Is this behavior by design or is it an oversight**?

If it is by design then:
1. What would be the use cases for having `dragstart` fired globally?
2. I believe this behavior should be more clearly stated in the README.
3. What can I do to guarantee that, for example, dragging component 'one' will call only its own `dragstart` handler and not all the others as well?

**Comment by rlemaigre** [comment]
No it is a mistake.

I have less and less time to work on this and issues are piling up. Perhaps what I should do is, properly document how this whole thing work, which is not that complicated, so that people can figure out what's wrong with the code and submit pull requests...

**Comment by rlemaigre** [comment]
The mistake is here : https://github.com/rlemaigre/Easy-DnD/blob/65a19bb8b915dd72a86befba032be2c831ab4e76/lib/src/mixins/DragMixin.ts#L35

I should only emit dragstart and dragend events if the source of the drag (which is referenced in the event) is the current drag component.

**Comment by rlemaigre** [comment]
Fixed in 1.8.3
